src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs

```diff
@@ -341,15 +341,16 @@ public async Task GetTemplateCollectionFromAcr_WhenGivenHl7v2DataForConverting_I
 
         [Theory]
         [MemberData(nameof(GetDefaultTemplatesInfo))]
-        public async Task GiveDefaultImageReference_WhenGetTemplateCollectionWithEmptyToken_DefaultTemplatesWillBeReturnedAsync(string imageReference, string expectedTemplatesFolder)
+        public async Task GiveDefaultImageReference_WhenGetTemplateCollectionWithEmptyToken_DefaultTemplatesWillBeReturnedAsync(string imageReference, int expectedTemplatesCount)
         {
             TemplateCollectionProviderFactory factory = new TemplateCollectionProviderFactory(cache, Options.Create(_config));
             var templateCollectionProvider = factory.CreateTemplateCollectionProvider(imageReference, string.Empty);
             var templateCollection = await templateCollectionProvider.GetTemplateCollectionAsync();
             Assert.Single(templateCollection);
 
-            // metadata.json will not be returned as templates.
-            Assert.Equal(Directory.GetFiles(Path.Join(_templateDirectory, expectedTemplatesFolder), "*", SearchOption.AllDirectories).Length - 1, templateCollection.First().Count());
+            // metadata.json will not be returned as template.
+            // Json/Schema/meta-schema.json will not be returned as template.
+            Assert.Equal(expectedTemplatesCount, templateCollection.First().Count());
```

Comment:
If we use the fixed number, I think we can consider such a condition `Assert.True(expectedTemplatesCounts <= templateCollection.First().Count());`

Reply:
Updated to check the files count.